src/sentry/integrations/slack/requests/base.py

```diff
@@ -120,17 +123,22 @@ def logging_data(self) -> Mapping[str, str]:
 
         return {k: v for k, v in data.items() if v}
 
-    def get_identity(self) -> Identity | None:
-        try:
-            idp = IdentityProvider.objects.get(type="slack", external_id=self.team_id)
-        except IdentityProvider.DoesNotExist as e:
-            logger.info("slack.action.invalid-team-id", extra={"slack_team": self.team_id})
-            raise e
-
-        try:
-            return Identity.objects.select_related("user").get(idp=idp, external_id=self.user_id)
-        except Identity.DoesNotExist:
+    def get_identity(self) -> APIIdentity | None:
+        provider = identity_service.get_provider(
+            provider_type="slack", provider_ext_id=self.team_id
+        )
+        self._identity = (
+            identity_service.get_identity(provider_id=provider.id, identity_ext_id=self.user_id)
+            if provider
+            else None
+        )
+        return self._identity
+
+    def get_identity_user(self) -> APIUser | None:
+        identity = self._identity if self._identity else self.get_identity()
```

Comment:
This can just become `identity = self.get_identity()` if `get_identity` is made fully idempotent.

Reply:
makes sense :+1: